fe/fe-core/src/main/java/org/apache/doris/catalog/HMSResource.java

```diff
@@ -97,7 +97,33 @@ protected void getProcNodeData(BaseProcResult result) {
         }
     }
 
+    private static void getPropertiesFromDLFProps(Map<String, String> props) {
+        // add transformed properties
+        String propsMetastoreType = props.get(HIVE_METASTORE_TYPE);
+        if (!DLF_TYPE.equalsIgnoreCase(propsMetastoreType)) {
+            return;
+        }
+        props.put(S3Resource.S3_ACCESS_KEY, props.getOrDefault("dlf.catalog.accessKeyId", ""));
+        props.put(S3Resource.S3_SECRET_KEY, props.getOrDefault("dlf.catalog.accessKeySecret", ""));
+        String region = props.get("dlf.catalog.region");
+        if (!Strings.isNullOrEmpty(region)) {
+            props.put(S3Resource.S3_REGION, "oss-" + region);
+            String publicAccess = props.getOrDefault("dlf.catalog.accessPublic", "false");
+            props.put(S3Resource.S3_ENDPOINT, getDLFEndpont(region, Boolean.parseBoolean(publicAccess)));
+        }
+    }
+
+    private static String getDLFEndpont(String region, boolean publicAccess) {
+        String prefix = "http://oss-";
+        String suffix = ".aliyuncs.com";
+        if (!publicAccess) {
+            suffix = "-internal" + suffix;
+        }
+        return prefix + region + suffix;
+    }
+
     public static Map<String, String> getPropertiesFromDLF(Map<String, String> res) {
+        getPropertiesFromDLFProps(res);
```

Comment:
The priority of properties in `create catalog` stmt is higher than properties in `hive-site.xml`.
So you need first read props from file, than replace it with properties in stmt.

Reply:
The priority from high to low is:
1. properties in create catalog stmt
2. properties in resource
3. properties in xml